WordPress/gutenberg
# Duotone: Fix CSS Selectors rendered by theme.json duotone/filter settings for blocks on public pages

## What?

This PR fixes the selectors rendered in the CSS generated by the `theme.json` filter/duotone settings for block types. 
The issue is described here: https://github.com/WordPress/gutenberg/issues/41333

## Why?

Currently, we are rendering all the CSS generated on the server based on `theme.json` block styles pointing to a single CSS selector targeting the main block container element. This situation causes the problems described in the issue. This change filters the definition to render the CSS regarding filter to the selector defined by the block definition.

The fix is necessary to avoid:
- Discrepancies between what is shown in the editor and the public version of the post.
- Overriding the default filter/duotone using the editor has an unexpected result.

## How?

The change introduced here filters the CSS rules to be able to write different CSS selectors. A selector for the main block container and a selector for the duotone element indicated in the block definition (`block.json`).

## Testing Instructions





1. Use Skatepark or another theme with default duotone filters applied to some blocks such as Image and Cover.
2. Create a new post and add an Image block, and also a Cover Block.
3. Select a different duotone filter for each of those blocks recently added.
4. Save and see the public version of the post.


## Screenshots or screencast 

**Before** (left: editor, right: public post):
![image](https://user-images.githubusercontent.com/1310626/170296733-bee639ae-f217-48d3-ab48-2b6830f40d01.png)


**After** (left: editor, right: public post):
![image](https://user-images.githubusercontent.com/1310626/170296902-5e8cbe72-a814-4800-8bcf-6721dfd46aa9.png)
